components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1256,17 +1256,20 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
     }
 
     async getEnvVars(): Promise<UserEnvVarValue[]> {
```

Comment:
This made me think: Would it make sense to have the filtering for repositoryPattern on the server/API side?